src/date-picker/__tests__/date-picker-calendar.test.tsx

```diff
@@ -435,7 +437,10 @@ describe('Date picker calendar', () => {
         const { wrapper } = renderDatePicker({ ...defaultProps, value: '2018-03-21', isDateEnabled });
         act(() => wrapper.findOpenCalendarButton().click());
         expect(findCalendarHeaderText(wrapper)).toBe('March 2018');
-        expect(wrapper.findCalendar()!.findSelectedDate().getElement().textContent).toBe('21');
+        expect(
+          wrapper.findCalendar()!.findSelectedDate()?.find(`:not(.${calendarStyles['visually-hidden']})`)?.getElement()
```

Comment:
The `findSelectedDate` is a public selector. We need to exclude the hidden content as part of it to not break the customer code.